asyncawait fails to build on Ubuntu 16.04.4 LTS

Hello. I am completely blocked as asyncawait fails to build on Ubuntu 16.04.4 

Can anyone help resolve the core issue?

Build Errors below
<pre>
spartanlync-mobile!valor [/var/valor/temp/asyncawait]>npm install -g

> fibers@1.0.15 install /var/valor/temp/asyncawait/node_modules/fibers
> node build.js || nodejs build.js

make: Entering directory '/var/valor/temp/asyncawait/node_modules/fibers/build'
  CXX(target) Release/obj.target/fibers/src/fibers.o
../src/fibers.cc: In function ‘v8::Handle<v8::String> uni::NewLatin1String(v8::Isolate*, const char*)’:
../src/fibers.cc:122:64: error: no matching function for call to ‘v8::String::NewFromOneByte(v8::Isolate*&, const uint8_t*)’
   return String::NewFromOneByte(isolate, (const uint8_t*)string);
                                                                ^
In file included from /home/valor/.node-gyp/10.0.0/include/node/node.h:63:0,
                 from ../src/coroutine.h:1,
                 from ../src/fibers.cc:1:
/home/valor/.node-gyp/10.0.0/include/node/v8.h:2736:51: note: candidate: static v8::MaybeLocal<v8::String> v8::String::NewFromOneByte(v8::Isolate*, const uint8_t*, v8::NewStringType, int)
   static V8_WARN_UNUSED_RESULT MaybeLocal<String> NewFromOneByte(
                                                   ^
/home/valor/.node-gyp/10.0.0/include/node/v8.h:2736:51: note:   candidate expects 4 arguments, 2 provided
../src/fibers.cc: In function ‘v8::Handle<v8::String> uni::NewLatin1Symbol(v8::Isolate*, const char*)’:
../src/fibers.cc:126:64: error: no matching function for call to ‘v8::String::NewFromOneByte(v8::Isolate*&, const uint8_t*)’
   return String::NewFromOneByte(isolate, (const uint8_t*)string);
                                                                ^
In file included from /home/valor/.node-gyp/10.0.0/include/node/node.h:63:0,
                 from ../src/coroutine.h:1,
                 from ../src/fibers.cc:1:
/home/valor/.node-gyp/10.0.0/include/node/v8.h:2736:51: note: candidate: static v8::MaybeLocal<v8::String> v8::String::NewFromOneByte(v8::Isolate*, const uint8_t*, v8::NewStringType, int)
   static V8_WARN_UNUSED_RESULT MaybeLocal<String> NewFromOneByte(
                                                   ^
/home/valor/.node-gyp/10.0.0/include/node/v8.h:2736:51: note:   candidate expects 4 arguments, 2 provided
../src/fibers.cc: In static member function ‘static void Fiber::DestroyOrphans()’:
../src/fibers.cc:406:67: warning: ‘v8::String::Utf8Value::Utf8Value(v8::Local<v8::Value>)’ is deprecated: Use Isolate version [-Wdeprecated-declarations]
      String::Utf8Value stack(uni::Deref(that.isolate, fatal_stack));
                                                                   ^
In file included from /home/valor/.node-gyp/10.0.0/include/node/v8.h:26:0,
                 from /home/valor/.node-gyp/10.0.0/include/node/node.h:63,
                 from ../src/coroutine.h:1,
                 from ../src/fibers.cc:1:
/home/valor/.node-gyp/10.0.0/include/node/v8.h:2822:28: note: declared here
                   explicit Utf8Value(Local<v8::Value> obj));
                            ^
/home/valor/.node-gyp/10.0.0/include/node/v8config.h:318:3: note: in definition of macro ‘V8_DEPRECATED’
   declarator __attribute__((deprecated(message)))
   ^
../src/fibers.cc: In static member function ‘static uni::FunctionType Fiber::New(const Arguments&)’:
../src/fibers.cc:433:99: error: no matching function for call to ‘v8::Function::NewInstance(int, v8::Handle<v8::Value> [1])’
     return uni::Return(uni::Deref(Isolate::GetCurrent(), tmpl)->GetFunction()->NewInstance(1, argv), args);
                                                                                                   ^
In file included from /home/valor/.node-gyp/10.0.0/include/node/node.h:63:0,
                 from ../src/coroutine.h:1,
                 from ../src/fibers.cc:1:
/home/valor/.node-gyp/10.0.0/include/node/v8.h:3848:44: note: candidate: v8::MaybeLocal<v8::Object> v8::Function::NewInstance(v8::Local<v8::Context>, int, v8::Local<v8::Value>*) const
   V8_WARN_UNUSED_RESULT MaybeLocal<Object> NewInstance(
                                            ^
/home/valor/.node-gyp/10.0.0/include/node/v8.h:3848:44: note:   candidate expects 3 arguments, 2 provided
/home/valor/.node-gyp/10.0.0/include/node/v8.h:3851:44: note: candidate: v8::MaybeLocal<v8::Object> v8::Function::NewInstance(v8::Local<v8::Context>) const
   V8_WARN_UNUSED_RESULT MaybeLocal<Object> NewInstance(
                                            ^
/home/valor/.node-gyp/10.0.0/include/node/v8.h:3851:44: note:   candidate expects 1 argument, 2 provided
../src/fibers.cc:433:106: error: return-statement with a value, in function returning 'void' [-fpermissive]
     return uni::Return(uni::Deref(Isolate::GetCurrent(), tmpl)->GetFunction()->NewInstance(1, argv), args);
                                                                                                          ^
../src/fibers.cc: In static member function ‘static void Fiber::RunFiber(void**)’:
../src/fibers.cc:621:14: error: no matching function for call to ‘v8::TryCatch::TryCatch()’
     TryCatch try_catch;
              ^
In file included from /home/valor/.node-gyp/10.0.0/include/node/node.h:63:0,
                 from ../src/coroutine.h:1,
                 from ../src/fibers.cc:1:
/home/valor/.node-gyp/10.0.0/include/node/v8.h:8271:12: note: candidate: v8::TryCatch::TryCatch(v8::Isolate*)
   explicit TryCatch(Isolate* isolate);
            ^
/home/valor/.node-gyp/10.0.0/include/node/v8.h:8271:12: note:   candidate expects 1 argument, 0 provided
../src/fibers.cc:628:66: warning: ‘static v8::Local<v8::Script> v8::Script::Compile(v8::Local<v8::String>, v8::ScriptOrigin*)’ is deprecated: Use maybe version [-Wdeprecated-declarations]
     Script::Compile(uni::NewLatin1String(that.isolate, "void 0;"));
                                                                  ^
In file included from /home/valor/.node-gyp/10.0.0/include/node/v8.h:26:0,
                 from /home/valor/.node-gyp/10.0.0/include/node/node.h:63,
                 from ../src/coroutine.h:1,
                 from ../src/fibers.cc:1:
/home/valor/.node-gyp/10.0.0/include/node/v8.h:1235:38: note: declared here
                        Local<Script> Compile(Local<String> source,
                                      ^
/home/valor/.node-gyp/10.0.0/include/node/v8config.h:318:3: note: in definition of macro ‘V8_DEPRECATED’
   declarator __attribute__((deprecated(message)))
   ^
../src/fibers.cc:643:66: warning: ‘v8::Local<v8::Value> v8::TryCatch::StackTrace() const’ is deprecated: Use maybe version. [-Wdeprecated-declarations]
       uni::Reset(that.isolate, fatal_stack, try_catch.StackTrace());
                                                                  ^
In file included from /home/valor/.node-gyp/10.0.0/include/node/v8.h:26:0,
                 from /home/valor/.node-gyp/10.0.0/include/node/node.h:63,
                 from ../src/coroutine.h:1,
                 from ../src/fibers.cc:1:
/home/valor/.node-gyp/10.0.0/include/node/v8.h:8328:52: note: declared here
   V8_DEPRECATED("Use maybe version.", Local<Value> StackTrace() const);
                                                    ^
/home/valor/.node-gyp/10.0.0/include/node/v8config.h:318:3: note: in definition of macro ‘V8_DEPRECATED’
   declarator __attribute__((deprecated(message)))
   ^
../src/fibers.cc: In static member function ‘static void Fiber::SetPoolSize(v8::Local<v8::String>, v8::Local<v8::Value>, const SetterCallbackInfo&)’:
../src/fibers.cc:741:43: error: no matching function for call to ‘v8::Value::ToNumber()’
    Coroutine::pool_size = value->ToNumber()->Value();
                                           ^
In file included from /home/valor/.node-gyp/10.0.0/include/node/node.h:63:0,
                 from ../src/coroutine.h:1,
                 from ../src/fibers.cc:1:
/home/valor/.node-gyp/10.0.0/include/node/v8.h:2366:44: note: candidate: v8::MaybeLocal<v8::Number> v8::Value::ToNumber(v8::Local<v8::Context>) const
   V8_WARN_UNUSED_RESULT MaybeLocal<Number> ToNumber(
                                            ^
/home/valor/.node-gyp/10.0.0/include/node/v8.h:2366:44: note:   candidate expects 1 argument, 0 provided
In file included from /home/valor/.node-gyp/10.0.0/include/node/v8.h:26:0,
                 from /home/valor/.node-gyp/10.0.0/include/node/node.h:63,
                 from ../src/coroutine.h:1,
                 from ../src/fibers.cc:1:
/home/valor/.node-gyp/10.0.0/include/node/v8.h:2383:35: note: candidate: v8::Local<v8::Number> v8::Value::ToNumber(v8::Isolate*) const
                     Local<Number> ToNumber(Isolate* isolate) const);
                                   ^
/home/valor/.node-gyp/10.0.0/include/node/v8config.h:340:48: note: in definition of macro ‘V8_DEPRECATE_SOON’
 #define V8_DEPRECATE_SOON(message, declarator) declarator
                                                ^
/home/valor/.node-gyp/10.0.0/include/node/v8.h:2383:35: note:   candidate expects 1 argument, 0 provided
                     Local<Number> ToNumber(Isolate* isolate) const);
                                   ^
/home/valor/.node-gyp/10.0.0/include/node/v8config.h:340:48: note: in definition of macro ‘V8_DEPRECATE_SOON’
 #define V8_DEPRECATE_SOON(message, declarator) declarator
                                                ^
../src/fibers.cc: In static member function ‘static void Fiber::Init(v8::Handle<v8::Object>)’:
../src/fibers.cc:793:72: error: no matching function for call to ‘v8::Function::SetAccessor(v8::Handle<v8::String>, uni::FunctionType (&)(v8::Local<v8::String>, const GetterCallbackInfo&))’
    fn->SetAccessor(uni::NewLatin1Symbol(isolate, "current"), GetCurrent);
                                                                        ^
In file included from /home/valor/.node-gyp/10.0.0/include/node/node.h:63:0,
                 from ../src/coroutine.h:1,
                 from ../src/fibers.cc:1:
/home/valor/.node-gyp/10.0.0/include/node/v8.h:3214:37: note: candidate: v8::Maybe<bool> v8::Object::SetAccessor(v8::Local<v8::Context>, v8::Local<v8::Name>, v8::AccessorNameGetterCallback, v8::AccessorNameSetterCallback, v8::MaybeLocal<v8::Value>, v8::AccessControl, v8::PropertyAttribute, v8::SideEffectType)
   V8_WARN_UNUSED_RESULT Maybe<bool> SetAccessor(
                                     ^
/home/valor/.node-gyp/10.0.0/include/node/v8.h:3214:37: note:   candidate expects 8 arguments, 2 provided
../src/fibers.cc:794:87: error: no matching function for call to ‘v8::Function::SetAccessor(v8::Handle<v8::String>, uni::FunctionType (&)(v8::Local<v8::String>, const GetterCallbackInfo&), void (&)(v8::Local<v8::String>, v8::Local<v8::Value>, const SetterCallbackInfo&))’
    fn->SetAccessor(uni::NewLatin1Symbol(isolate, "poolSize"), GetPoolSize, SetPoolSize);
                                                                                       ^
In file included from /home/valor/.node-gyp/10.0.0/include/node/node.h:63:0,
                 from ../src/coroutine.h:1,
                 from ../src/fibers.cc:1:
/home/valor/.node-gyp/10.0.0/include/node/v8.h:3214:37: note: candidate: v8::Maybe<bool> v8::Object::SetAccessor(v8::Local<v8::Context>, v8::Local<v8::Name>, v8::AccessorNameGetterCallback, v8::AccessorNameSetterCallback, v8::MaybeLocal<v8::Value>, v8::AccessControl, v8::PropertyAttribute, v8::SideEffectType)
   V8_WARN_UNUSED_RESULT Maybe<bool> SetAccessor(
                                     ^
/home/valor/.node-gyp/10.0.0/include/node/v8.h:3214:37: note:   no known conversion for argument 2 from ‘uni::FunctionType(v8::Local<v8::String>, const GetterCallbackInfo&) {aka void(v8::Local<v8::String>, const v8::PropertyCallbackInfo<v8::Value>&)}’ to ‘v8::Local<v8::Name>’
../src/fibers.cc:795:84: error: no matching function for call to ‘v8::Function::SetAccessor(v8::Handle<v8::String>, uni::FunctionType (&)(v8::Local<v8::String>, const GetterCallbackInfo&))’
    fn->SetAccessor(uni::NewLatin1Symbol(isolate, "fibersCreated"), GetFibersCreated);
                                                                                    ^
In file included from /home/valor/.node-gyp/10.0.0/include/node/node.h:63:0,
                 from ../src/coroutine.h:1,
                 from ../src/fibers.cc:1:
/home/valor/.node-gyp/10.0.0/include/node/v8.h:3214:37: note: candidate: v8::Maybe<bool> v8::Object::SetAccessor(v8::Local<v8::Context>, v8::Local<v8::Name>, v8::AccessorNameGetterCallback, v8::AccessorNameSetterCallback, v8::MaybeLocal<v8::Value>, v8::AccessControl, v8::PropertyAttribute, v8::SideEffectType)
   V8_WARN_UNUSED_RESULT Maybe<bool> SetAccessor(
                                     ^
/home/valor/.node-gyp/10.0.0/include/node/v8.h:3214:37: note:   candidate expects 8 arguments, 2 provided
fibers.target.mk:102: recipe for target 'Release/obj.target/fibers/src/fibers.o' failed
make: *** [Release/obj.target/fibers/src/fibers.o] Error 1
make: Leaving directory '/var/valor/temp/asyncawait/node_modules/fibers/build'
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:258:23)
gyp ERR! stack     at ChildProcess.emit (events.js:182:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:225:12)
gyp ERR! System Linux 4.4.0-124-generic
gyp ERR! command "/usr/bin/node" "/usr/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild" "--release"
gyp ERR! cwd /var/valor/temp/asyncawait/node_modules/fibers
gyp ERR! node -v v10.0.0
gyp ERR! node-gyp -v v3.6.2
gyp ERR! not ok 
node-gyp exited with code: 1
Please make sure you are using a supported platform and node version. If you
would like to compile fibers on this machine please make sure you have setup your
build environment--
Windows + OS X instructions here: https://github.com/nodejs/node-gyp
Ubuntu users please run: `sudo apt-get install g++ build-essential`
Alpine users please run: `sudo apk add python make g++`
make: Entering directory '/var/valor/temp/asyncawait/node_modules/fibers/build'
  CXX(target) Release/obj.target/fibers/src/fibers.o
../src/fibers.cc: In function ‘v8::Handle<v8::String> uni::NewLatin1String(v8::Isolate*, const char*)’:
../src/fibers.cc:122:64: error: no matching function for call to ‘v8::String::NewFromOneByte(v8::Isolate*&, const uint8_t*)’
   return String::NewFromOneByte(isolate, (const uint8_t*)string);
                                                                ^
In file included from /home/valor/.node-gyp/10.0.0/include/node/node.h:63:0,
                 from ../src/coroutine.h:1,
                 from ../src/fibers.cc:1:
/home/valor/.node-gyp/10.0.0/include/node/v8.h:2736:51: note: candidate: static v8::MaybeLocal<v8::String> v8::String::NewFromOneByte(v8::Isolate*, const uint8_t*, v8::NewStringType, int)
   static V8_WARN_UNUSED_RESULT MaybeLocal<String> NewFromOneByte(
                                                   ^
/home/valor/.node-gyp/10.0.0/include/node/v8.h:2736:51: note:   candidate expects 4 arguments, 2 provided
../src/fibers.cc: In function ‘v8::Handle<v8::String> uni::NewLatin1Symbol(v8::Isolate*, const char*)’:
../src/fibers.cc:126:64: error: no matching function for call to ‘v8::String::NewFromOneByte(v8::Isolate*&, const uint8_t*)’
   return String::NewFromOneByte(isolate, (const uint8_t*)string);
                                                                ^
In file included from /home/valor/.node-gyp/10.0.0/include/node/node.h:63:0,
                 from ../src/coroutine.h:1,
                 from ../src/fibers.cc:1:
/home/valor/.node-gyp/10.0.0/include/node/v8.h:2736:51: note: candidate: static v8::MaybeLocal<v8::String> v8::String::NewFromOneByte(v8::Isolate*, const uint8_t*, v8::NewStringType, int)
   static V8_WARN_UNUSED_RESULT MaybeLocal<String> NewFromOneByte(
                                                   ^
/home/valor/.node-gyp/10.0.0/include/node/v8.h:2736:51: note:   candidate expects 4 arguments, 2 provided
../src/fibers.cc: In static member function ‘static void Fiber::DestroyOrphans()’:
../src/fibers.cc:406:67: warning: ‘v8::String::Utf8Value::Utf8Value(v8::Local<v8::Value>)’ is deprecated: Use Isolate version [-Wdeprecated-declarations]
      String::Utf8Value stack(uni::Deref(that.isolate, fatal_stack));
                                                                   ^
In file included from /home/valor/.node-gyp/10.0.0/include/node/v8.h:26:0,
                 from /home/valor/.node-gyp/10.0.0/include/node/node.h:63,
                 from ../src/coroutine.h:1,
                 from ../src/fibers.cc:1:
/home/valor/.node-gyp/10.0.0/include/node/v8.h:2822:28: note: declared here
                   explicit Utf8Value(Local<v8::Value> obj));
                            ^
/home/valor/.node-gyp/10.0.0/include/node/v8config.h:318:3: note: in definition of macro ‘V8_DEPRECATED’
   declarator __attribute__((deprecated(message)))
   ^
../src/fibers.cc: In static member function ‘static uni::FunctionType Fiber::New(const Arguments&)’:
../src/fibers.cc:433:99: error: no matching function for call to ‘v8::Function::NewInstance(int, v8::Handle<v8::Value> [1])’
     return uni::Return(uni::Deref(Isolate::GetCurrent(), tmpl)->GetFunction()->NewInstance(1, argv), args);
                                                                                                   ^
In file included from /home/valor/.node-gyp/10.0.0/include/node/node.h:63:0,
                 from ../src/coroutine.h:1,
                 from ../src/fibers.cc:1:
/home/valor/.node-gyp/10.0.0/include/node/v8.h:3848:44: note: candidate: v8::MaybeLocal<v8::Object> v8::Function::NewInstance(v8::Local<v8::Context>, int, v8::Local<v8::Value>*) const
   V8_WARN_UNUSED_RESULT MaybeLocal<Object> NewInstance(
                                            ^
/home/valor/.node-gyp/10.0.0/include/node/v8.h:3848:44: note:   candidate expects 3 arguments, 2 provided
/home/valor/.node-gyp/10.0.0/include/node/v8.h:3851:44: note: candidate: v8::MaybeLocal<v8::Object> v8::Function::NewInstance(v8::Local<v8::Context>) const
   V8_WARN_UNUSED_RESULT MaybeLocal<Object> NewInstance(
                                            ^
/home/valor/.node-gyp/10.0.0/include/node/v8.h:3851:44: note:   candidate expects 1 argument, 2 provided
../src/fibers.cc:433:106: error: return-statement with a value, in function returning 'void' [-fpermissive]
     return uni::Return(uni::Deref(Isolate::GetCurrent(), tmpl)->GetFunction()->NewInstance(1, argv), args);
                                                                                                          ^
../src/fibers.cc: In static member function ‘static void Fiber::RunFiber(void**)’:
../src/fibers.cc:621:14: error: no matching function for call to ‘v8::TryCatch::TryCatch()’
     TryCatch try_catch;
              ^
In file included from /home/valor/.node-gyp/10.0.0/include/node/node.h:63:0,
                 from ../src/coroutine.h:1,
                 from ../src/fibers.cc:1:
/home/valor/.node-gyp/10.0.0/include/node/v8.h:8271:12: note: candidate: v8::TryCatch::TryCatch(v8::Isolate*)
   explicit TryCatch(Isolate* isolate);
            ^
/home/valor/.node-gyp/10.0.0/include/node/v8.h:8271:12: note:   candidate expects 1 argument, 0 provided
../src/fibers.cc:628:66: warning: ‘static v8::Local<v8::Script> v8::Script::Compile(v8::Local<v8::String>, v8::ScriptOrigin*)’ is deprecated: Use maybe version [-Wdeprecated-declarations]
     Script::Compile(uni::NewLatin1String(that.isolate, "void 0;"));
                                                                  ^
In file included from /home/valor/.node-gyp/10.0.0/include/node/v8.h:26:0,
                 from /home/valor/.node-gyp/10.0.0/include/node/node.h:63,
                 from ../src/coroutine.h:1,
                 from ../src/fibers.cc:1:
/home/valor/.node-gyp/10.0.0/include/node/v8.h:1235:38: note: declared here
                        Local<Script> Compile(Local<String> source,
                                      ^
/home/valor/.node-gyp/10.0.0/include/node/v8config.h:318:3: note: in definition of macro ‘V8_DEPRECATED’
   declarator __attribute__((deprecated(message)))
   ^
../src/fibers.cc:643:66: warning: ‘v8::Local<v8::Value> v8::TryCatch::StackTrace() const’ is deprecated: Use maybe version. [-Wdeprecated-declarations]
       uni::Reset(that.isolate, fatal_stack, try_catch.StackTrace());
                                                                  ^
In file included from /home/valor/.node-gyp/10.0.0/include/node/v8.h:26:0,
                 from /home/valor/.node-gyp/10.0.0/include/node/node.h:63,
                 from ../src/coroutine.h:1,
                 from ../src/fibers.cc:1:
/home/valor/.node-gyp/10.0.0/include/node/v8.h:8328:52: note: declared here
   V8_DEPRECATED("Use maybe version.", Local<Value> StackTrace() const);
                                                    ^
/home/valor/.node-gyp/10.0.0/include/node/v8config.h:318:3: note: in definition of macro ‘V8_DEPRECATED’
   declarator __attribute__((deprecated(message)))
   ^
../src/fibers.cc: In static member function ‘static void Fiber::SetPoolSize(v8::Local<v8::String>, v8::Local<v8::Value>, const SetterCallbackInfo&)’:
../src/fibers.cc:741:43: error: no matching function for call to ‘v8::Value::ToNumber()’
    Coroutine::pool_size = value->ToNumber()->Value();
                                           ^
In file included from /home/valor/.node-gyp/10.0.0/include/node/node.h:63:0,
                 from ../src/coroutine.h:1,
                 from ../src/fibers.cc:1:
/home/valor/.node-gyp/10.0.0/include/node/v8.h:2366:44: note: candidate: v8::MaybeLocal<v8::Number> v8::Value::ToNumber(v8::Local<v8::Context>) const
   V8_WARN_UNUSED_RESULT MaybeLocal<Number> ToNumber(
                                            ^
/home/valor/.node-gyp/10.0.0/include/node/v8.h:2366:44: note:   candidate expects 1 argument, 0 provided
In file included from /home/valor/.node-gyp/10.0.0/include/node/v8.h:26:0,
                 from /home/valor/.node-gyp/10.0.0/include/node/node.h:63,
                 from ../src/coroutine.h:1,
                 from ../src/fibers.cc:1:
/home/valor/.node-gyp/10.0.0/include/node/v8.h:2383:35: note: candidate: v8::Local<v8::Number> v8::Value::ToNumber(v8::Isolate*) const
                     Local<Number> ToNumber(Isolate* isolate) const);
                                   ^
/home/valor/.node-gyp/10.0.0/include/node/v8config.h:340:48: note: in definition of macro ‘V8_DEPRECATE_SOON’
 #define V8_DEPRECATE_SOON(message, declarator) declarator
                                                ^
/home/valor/.node-gyp/10.0.0/include/node/v8.h:2383:35: note:   candidate expects 1 argument, 0 provided
                     Local<Number> ToNumber(Isolate* isolate) const);
                                   ^
/home/valor/.node-gyp/10.0.0/include/node/v8config.h:340:48: note: in definition of macro ‘V8_DEPRECATE_SOON’
 #define V8_DEPRECATE_SOON(message, declarator) declarator
                                                ^
../src/fibers.cc: In static member function ‘static void Fiber::Init(v8::Handle<v8::Object>)’:
../src/fibers.cc:793:72: error: no matching function for call to ‘v8::Function::SetAccessor(v8::Handle<v8::String>, uni::FunctionType (&)(v8::Local<v8::String>, const GetterCallbackInfo&))’
    fn->SetAccessor(uni::NewLatin1Symbol(isolate, "current"), GetCurrent);
                                                                        ^
In file included from /home/valor/.node-gyp/10.0.0/include/node/node.h:63:0,
                 from ../src/coroutine.h:1,
                 from ../src/fibers.cc:1:
/home/valor/.node-gyp/10.0.0/include/node/v8.h:3214:37: note: candidate: v8::Maybe<bool> v8::Object::SetAccessor(v8::Local<v8::Context>, v8::Local<v8::Name>, v8::AccessorNameGetterCallback, v8::AccessorNameSetterCallback, v8::MaybeLocal<v8::Value>, v8::AccessControl, v8::PropertyAttribute, v8::SideEffectType)
   V8_WARN_UNUSED_RESULT Maybe<bool> SetAccessor(
                                     ^
/home/valor/.node-gyp/10.0.0/include/node/v8.h:3214:37: note:   candidate expects 8 arguments, 2 provided
../src/fibers.cc:794:87: error: no matching function for call to ‘v8::Function::SetAccessor(v8::Handle<v8::String>, uni::FunctionType (&)(v8::Local<v8::String>, const GetterCallbackInfo&), void (&)(v8::Local<v8::String>, v8::Local<v8::Value>, const SetterCallbackInfo&))’
    fn->SetAccessor(uni::NewLatin1Symbol(isolate, "poolSize"), GetPoolSize, SetPoolSize);
                                                                                       ^
In file included from /home/valor/.node-gyp/10.0.0/include/node/node.h:63:0,
                 from ../src/coroutine.h:1,
                 from ../src/fibers.cc:1:
/home/valor/.node-gyp/10.0.0/include/node/v8.h:3214:37: note: candidate: v8::Maybe<bool> v8::Object::SetAccessor(v8::Local<v8::Context>, v8::Local<v8::Name>, v8::AccessorNameGetterCallback, v8::AccessorNameSetterCallback, v8::MaybeLocal<v8::Value>, v8::AccessControl, v8::PropertyAttribute, v8::SideEffectType)
   V8_WARN_UNUSED_RESULT Maybe<bool> SetAccessor(
                                     ^
/home/valor/.node-gyp/10.0.0/include/node/v8.h:3214:37: note:   no known conversion for argument 2 from ‘uni::FunctionType(v8::Local<v8::String>, const GetterCallbackInfo&) {aka void(v8::Local<v8::String>, const v8::PropertyCallbackInfo<v8::Value>&)}’ to ‘v8::Local<v8::Name>’
../src/fibers.cc:795:84: error: no matching function for call to ‘v8::Function::SetAccessor(v8::Handle<v8::String>, uni::FunctionType (&)(v8::Local<v8::String>, const GetterCallbackInfo&))’
    fn->SetAccessor(uni::NewLatin1Symbol(isolate, "fibersCreated"), GetFibersCreated);
                                                                                    ^
In file included from /home/valor/.node-gyp/10.0.0/include/node/node.h:63:0,
                 from ../src/coroutine.h:1,
                 from ../src/fibers.cc:1:
/home/valor/.node-gyp/10.0.0/include/node/v8.h:3214:37: note: candidate: v8::Maybe<bool> v8::Object::SetAccessor(v8::Local<v8::Context>, v8::Local<v8::Name>, v8::AccessorNameGetterCallback, v8::AccessorNameSetterCallback, v8::MaybeLocal<v8::Value>, v8::AccessControl, v8::PropertyAttribute, v8::SideEffectType)
   V8_WARN_UNUSED_RESULT Maybe<bool> SetAccessor(
                                     ^
/home/valor/.node-gyp/10.0.0/include/node/v8.h:3214:37: note:   candidate expects 8 arguments, 2 provided
fibers.target.mk:102: recipe for target 'Release/obj.target/fibers/src/fibers.o' failed
make: *** [Release/obj.target/fibers/src/fibers.o] Error 1
make: Leaving directory '/var/valor/temp/asyncawait/node_modules/fibers/build'
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:258:23)
gyp ERR! stack     at ChildProcess.emit (events.js:182:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:225:12)
gyp ERR! System Linux 4.4.0-124-generic
gyp ERR! command "/usr/bin/node" "/usr/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild" "--release"
gyp ERR! cwd /var/valor/temp/asyncawait/node_modules/fibers
gyp ERR! node -v v10.0.0
gyp ERR! node-gyp -v v3.6.2
gyp ERR! not ok 
node-gyp exited with code: 1
Please make sure you are using a supported platform and node version. If you
would like to compile fibers on this machine please make sure you have setup your
build environment--
Windows + OS X instructions here: https://github.com/nodejs/node-gyp
Ubuntu users please run: `sudo apt-get install g++ build-essential`
Alpine users please run: `sudo apk add python make g++`
npm WARN rollback Rolling back fibers@1.0.15 failed (this is probably harmless): /var/valor/temp/asyncawait/node_modules/fibers is not a child of /usr/lib
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! fibers@1.0.15 install: `node build.js || nodejs build.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the fibers@1.0.15 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
</pre>

--- a/package.json
+++ b/package.json
@@ -42,15 +42,15 @@
   },
   "dependencies": {
     "bluebird": "^3.1.1",
-    "fibers": "^1.0.8",
-    "lodash": "v3.10.1"
+    "fibers": "^2.0.2",
+    "lodash": "^4.17.10"
   },
   "devDependencies": {
     "async": "^1.4.2",
     "asyncx": "^0.4.5",
     "chai": "^3.2.0",
     "co": "^4.6.0",
-    "mocha": "^2.3.2",
+    "mocha": "^5.2.0",
     "mock-fs": "^3.1.0",
     "rewire": "^2.3.4",
     "typescript": "1.7.5"

